inspirehep/modules/multieditor/queries.py

```diff
@@ -0,0 +1,122 @@
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE; if not, write to the
+# Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+# MA 02111-1307, USA.
+#
+# In applying this license, CERN does not
+# waive the privileges and immunities granted to it by virtue of its status
+# as an Intergovernmental Organization or submit itself to any jurisdiction.
+
+"""Module for quering in multi record editor used in http://inspirehep.net."""
+from __future__ import absolute_import, print_function, division
+from ..search import api
+from invenio_records.api import Record
+
+
+def get_records_from_query(query_string, size, page, index):
+    if index == 'hep':
+        query_result = api.LiteratureSearch().query_from_iq(query_string).params(
+            size=size,
+            from_=((page - 1) * 10),
+            _source=['control_number']
+        ).execute()
+    elif index == 'authors':
+        query_result = api.AuthorsSearch().query_from_iq(query_string).params(
+            size=size,
+            from_=((page - 1) * 10),
+            _source=['control_number']
+        ).execute()
+    elif index == 'data':
+        query_result = api.DataSearch().query_from_iq(query_string).params(
+            size=size,
+            from_=((page - 1) * 10),
+            _source=['control_number']
+        ).execute()
+    elif index == 'conferences':
+        query_result = api.ConferencesSearch().query_from_iq(query_string).params(
+            size=size,
+            from_=((page - 1) * 10),
+            _source=['control_number']
+        ).execute()
+    elif index == 'jobs':
+        query_result = api.JobsSearch().query_from_iq(query_string).params(
+            size=size,
+            from_=((page - 1) * 10),
+            _source=['control_number']
+        ).execute()
+    elif index == 'institutions':
+        query_result = api.InstitutionsSearch().query_from_iq(query_string).params(
+            size=size,
+            from_=((page - 1) * 10),
+            _source=['control_number']
+        ).execute()
+    elif index == 'experiments':
+        query_result = api.ExperimentsSearch().query_from_iq(query_string).params(
+            size=size,
+            from_=((page - 1) * 10),
+            _source=['control_number']
+        ).execute()
+    elif index == 'journals':
+        query_result = api.JournalsSearch().query_from_iq(query_string).params(
+            size=size,
+            from_=((page - 1) * 10),
+            _source=['control_number']
+        ).execute()
+    ids = [q.meta.id for q in query_result]
+    db_records = Record.get_records(ids)
+    return {
+        'json_records': db_records,
+        'total_records': query_result.hits.total
+    }
+
+
+def get_record_ids_from_query(query_string, index):
+    if index == 'hep':
```

Comment:
Same as above to make it compact.

As we talked, here you are only going to get the default size of 10 ids in all queries.